### PR TITLE
Add Rust porting documentation.

### PR DESCRIPTION
**Pull request by shaleh**

### CODE REVIEW

File: rust_porting_cookbook.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+`lisp_fn`, like `#[lisp_fn(name = "+")]`.
+
+Optional arguments are also possible: to make the minimum number of
+arguments from elisp different from the number of Rust arguments,
+pass a `min = "n"` argument.
+
+The docstring of the function should be the same as the docstring
+in the C code.  (Don't wonder about it being a comment there, Emacs
+has some magic that extracts it into a separate file.)
+
```
Comment (DavidDeSimone) [inline rust_porting_cookbook.md]: Maybe include a mention of what to do if the function is interactive? That can be an easy thing to miss.

Comment (shaleh) [reply rust_porting_cookbook.md]: There is info later in the document. Is it not enough?

Comment (DavidDeSimone) [reply rust_porting_cookbook.md]: Ah I missed that. The info later in the document seems fine.

Comment (shaleh) [reply rust_porting_cookbook.md]: I could link to the section from this location if that would help the reader.

**Comment by shaleh** [comment]
As we get further down the path of porting to Rust I expect someone will be looking at some C code and wondering where it is in Rust and how to adapt the changes. I started this document to gather this information in one place.

**Comment by shaleh** [comment]
Right now it is in the top of the tree. This is probably not the best location.

File: rust_porting_cookbook.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+When writing a Rust version of a C function, give it the same name and
+same arguments. If this isn't appropriate, docstrings should say the
+equivalent C function to help future porters.
+
+For example, `make_natnum` mentions that it can be used
+in place of `XSETFASTINT`.
+
+### C Macros
+
+For C macros, we try to define a fairly equivalent Rust
```
Comment (shanavas786) [inline rust_porting_cookbook.md]: a single macro/group of macros in C are often replaced by rust macro. eg: xsignal, call, wrong_type_arg etc..

Comment (shaleh) [reply rust_porting_cookbook.md]: Agreed.

We were talking in Gitter channel last night about how this doc might work better as a Wiki page.

Comment (shanavas786) [reply rust_porting_cookbook.md]: Yeah. That would be better

**Comment by shaleh** [comment]
I opened #481 to track the conversation around having a wiki.

**Comment by shaleh** [comment]
Agreed. Will do.

**Comment by shaleh** [comment]
I am closing this PR in favor of the wiki for now.

**Pull request closed without merging**